### PR TITLE
feat!: use regbot for clinical trial fetching

### LINKED ISSUE #85: Capture Recruitment Status in get_clinical_trials()
### Feature description

Currently get_clinical_trials() does not capture recruiting status for clinical trials

We should could consider clearly capturing and enumerating these values for downstream applications. 

### Use case

It might be frustrating to receive clinical trials results but then find out from a secondary source that these are not something that a patient can be enrolled in.

### Acceptance Criteria

The recruitment values should be reported in the table.

### Proposed solution

_No response_

### Alternatives considered

_No response_

### Implementation details

_No response_

### Potential Impact

_No response_

### Additional context

_No response_

### Contribution

None

### LINKED ISSUE #86: Ensure pagination is functioning in get_clinical_trials()
### Describe the bug

`dgipy.get_clinical_trials() `retrieves a max of 10 results no matter the search term

### Steps to reproduce

```
>>> 
>>> import dgipy
>>> import pandas as pd
>>> pd.DataFrame(dgipy.get_clinical_trials(['IMATINIB'])
... )
  search_term  ...                                      interventions
0    IMATINIB  ...  {'armGroups': [{'label': 'Gleevec', 'type': 'E...
1    IMATINIB  ...  {'armGroups': [{'label': 'Antineoplastic Drugs...
2    IMATINIB  ...  {'armGroups': [{'label': 'Arm', 'type': 'OTHER...
3    IMATINIB  ...  {'armGroups': [{'label': 'imatinib mesylate', ...
4    IMATINIB  ...  {'armGroups': [{'label': 'Treatment', 'type': ...
5    IMATINIB  ...  {'armGroups': [{'label': 'CML patients with MM...
6    IMATINIB  ...  {'armGroups': [{'label': 'sunitinib', 'type': ...
7    IMATINIB  ...  {'armGroups': [{'label': 'Intervention', 'type...
8    IMATINIB  ...  {'interventions': [{'type': 'DRUG', 'name': 'e...
9    IMATINIB  ...  {'armGroups': [{'label': '1', 'type': 'EXPERIM...

[10 rows x 9 columns]
>>> pd.DataFrame(dgipy.get_clinical_trials(['ASPIRIN']))
  search_term  ...                                      interventions
0     ASPIRIN  ...  {'armGroups': [{'label': 'Medical Management',...
1     ASPIRIN  ...  {'armGroups': [{'label': 'Cloas', 'type': 'EXP...
2     ASPIRIN  ...  {'armGroups': [{'label': 'Group 1- PREloading ...
3     ASPIRIN  ...  {'armGroups': [{'label': 'A', 'type': 'NO_INTE...
4     ASPIRIN  ...  {'armGroups': [{'label': '1', 'type': 'EXPERIM...
5     ASPIRIN  ...  {'armGroups': [{'label': '1, POLYCAP', 'type':...
6     ASPIRIN  ...  {'armGroups': [{'label': 'A : Alteplase', 'typ...
7     ASPIRIN  ...  {'interventions': [{'type': 'DRUG', 'name': 'A...
8     ASPIRIN  ...  {'armGroups': [{'label': 'SR-ASA', 'type': 'EX...
9     ASPIRIN  ...  {'armGroups': [{'label': 'platelet administere...

[10 rows x 9 columns]
>>> 


```

### Expected behavior

I would expect a dataframe of variable size, with drastically more results for both of these drugs:
Screenshot 2024-11-25 at 9 38 07 AM

### Current behavior

10 results were shown when there should be drastically more data, see above.

### Acceptance Criteria

We can figure out how to filter / add options for sorting retrieved data, but for now acceptance would be if results reasonably match up with what we can grab from the UI at https://clinicaltrials.gov/search?intr=Imatinib

### Possible reason(s)

_No response_

### Suggested fix

_No response_

### Branch, commit, and/or version

Main: b9ece0667

### Screenshots

_No response_

### Environment details

Pyenv python 3.11, Mac Sonoma 14.3

### Additional details

_No response_

### Contribution

Yes, I can create a PR for this fix.

### PR DESCRIPTION
**Pull request by jsstevenson**
progress on #59 
close #86 
close #85

### CODE REVIEW

File: src/dgipy/integration/clinical_trials.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
+"""Integrate data from FDA clinical trials API."""
+
+from regbot.fetch.clinical_trials import StandardAge
+from regbot.fetch.clinical_trials import get_clinical_trials as get_trials_from_fda
+
+
+def get_clinical_trials(terms: list) -> dict:
```
Comment (korikuzma) [inline src/dgipy/integration/clinical_trials.py:7]: Maybe specify that items in the list should be a str?
```suggestion
def get_clinical_trials(terms: list[str]) -> dict:
```

**Review by korikuzma — APPROVED** [review]
Minor suggestions

File: src/dgipy/integration/clinical_trials.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    :param terms: drugs of interest
+    :return: all clinical trials data for drugs of interest in a DataFrame-ready dict
+    """
+    if isinstance(terms, str):
+        # we shouldn't be too picky about input types, but this is an easy mistake
+        # that's worth trying to catch
+        msg = (
+            "`get_clinical_trials()` takes a list of terms as a parameter, not a string"
+        )
+        raise ValueError(msg)
```
[the remark this answers is not in the document]
Comment (jsstevenson) [reply src/dgipy/integration/clinical_trials.py]: I don't necessarily want to force a list (any non-string sequence should be fine) but I just left it as a warning so it's not too constrictive

**Comment by jsstevenson** [comment]
also tagging @mcannon068nw

File: src/dgipy/integrations/clinical_trials.py [hunk 3]
```diff
@@ -2,12 +2,90 @@
...
+from regbot.fetch.clinical_trials import StandardAge, Status, Study
 from regbot.fetch.clinical_trials import get_clinical_trials as get_trials_from_fda
 
 _logger = logging.getLogger(__name__)
 
 
+def _add_study_to_output(output: dict[str, list], drug_name: str, study: Study) -> None:
+    """Update `output` in-place with results from study
+
+    :param output: in-progress raw columnar data
```
Comment (korikuzma) [inline src/dgipy/integrations/clinical_trials.py:14]: missing drug_name

**Review by korikuzma — APPROVED** [review]

**Review by korikuzma — APPROVED** [review]

**Pull request merged**